### PR TITLE
[GUITextLayout] Removed custom chinese chars line break

### LINKED ISSUE #25684: The edge text(Simplified Chinese) is blocked and incompletely displayed.
### Describe the bug
Here is a clear and concise description of what the problem is:
The edge text is blocked and incompletely displayed.

## Expected Behavior
Here is a clear and concise description of what was expected to happen:
kodi-21.0-Omega is ok:
kodi-21 0-Omega-p1
kodi-21 0-Omega-p2
kodi-21 0-Omega-p3

## Actual Behavior
Kodi-20240828 Nightly, The edge text is blocked and incompletely displayed.
Kodi-20240828-p1
Kodi-20240828-p2
Kodi-20240828-p3

### Your Environment
Used Operating system:
Android TV(android 11) & Windows(Win 11) has same problem
Kodi version: Kodi-20240828 Nightly
Kodi-20240828-info

### PR DESCRIPTION
**Pull request by CastagnaIT**
## Description

after the #24905 rework that has fixed text wrap and justify alignment
old code was breaking lines with a custom behaviour for chinese chars that i have not realized before
this is not needed with the new code

## Motivation and context

fix #25684
## How has this been tested?

user confirmed
## Screenshots (if appropriate):
BEFORE
immagine
AFTER, THE YELLOW CHARACTER, BEFORE MISSING, IS NOW SHOWN CORRECTLY
immagine

## Types of change

- [x] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **Student submission** (PR was done for educational purposes and will be treated as such)
- [ ] **None of the above** (please explain below)

## Checklist:

- [ ] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [ ] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

File: xbmc/guilib/GUITextLayout.h [hunk 1]
```diff
@@ -187,7 +187,7 @@ class CGUITextLayout
   inline bool CanWrapAtLetter(character_t letter) const XBMC_FORCE_INLINE
   {
     character_t ch = letter & 0xffff;
-    return ch == L' ' || (ch >=0x4e00 && ch <= 0x9fff);
+    return ch == L' ';
```
Comment (CrystalP) [inline xbmc/guilib/GUITextLayout.h:190]: The previous condition that included the "CJK Unified Ideographs" block is strange and dates back to the original xbox code. No problem with the removal, though we won't know why it was added.
However maybe add a todo comment that unicode spaces are not handled (yet) - category Zs or maybe attribute white space. If that happens one day, non-breaking space will require special handling.

Comment (CastagnaIT) [reply xbmc/guilib/GUITextLayout.h:190]: i can add the comment, anyway here its not the only place where would be needed handle also unicode spaces

Comment (CastagnaIT) [reply xbmc/guilib/GUITextLayout.h:191]: done

**Review by CrystalP — APPROVED** [review]
looks good.

**Comment by scott967** [comment]
A quick look at Japanese text and it does fix the immediate issue with dropping characters.  However, looking at GUITextLayout I Can't say we are doing a great job wrapping text.  For example, I see line break before IDEOGRAPHIC FULL STOP U+3002.

At some point it might be worth looking at Unicode UAX#14 line break algorithm.  It can be tailored for our use case which doesn't require support for every possible unicode script/language.

**Comment by CastagnaIT** [comment]
jenkins build this please

**Comment by CastagnaIT** [comment]
CI error not related to PR, merging..

**Pull request merged**